lib/loader/index.js

```diff
@@ -289,7 +290,17 @@ function demangle(exports, baseModule) {
           });
         }
       } else {
-        curr[name] = wrapFunction(elem, setargc);
+        if (name === 'constructor') {
+          curr[name] = wrapFunction(elem, setargc);
+        } else { // for methods
+          Object.defineProperty(curr, name, {
+            value: function (...args) {
+              args.unshift(this.this);
+              setargc(args.length);
+              return elem(...args);
```

Comment:
Similar: `return elem(this.this, ...args);`

Reply:
@dcodeIO `this.this` looks strange. May be better rename it to `self` or `value` for `wrap` function?

Reply:
I only used `.this` as the property name because is is less likely to conflict with a class's own fields and methods. Could still conflict, though.

Reply:
got it. What about using Symbol?

Reply:
Good call, done in https://github.com/AssemblyScript/assemblyscript/commit/86c084a519889d44f0f91ada09a3d2adaffb3d0a